instagram/src/components/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Comment from '../Comment/Comment';
+import { CommentSectionContainer } from './CommentSectionStyles';
+
+class CommentSection extends React.Component {
+  state = { comments: this.props.comments, newComment: '' };
+
+  addNewComment = (event) => {
+    event.preventDefault();
+    console.log('before function', this.state.comments);
+    if (this.state.newComment.length > 0) {
+      this.setState({
+        comments: [
+          ...this.state.comments,
+          {
+            username: localStorage.getItem('username'),
+            text: this.state.newComment,
+          },
+        ],
+        newComment: '',
+      });
+    }
+    console.log('after function', this.state.comments);
+  };
+
+  changeHandler = (event) => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  render() {
+    return (
+      <CommentSectionContainer>
+        {this.state.comments.map((comment) => {
+          return <Comment comment={comment} key={comment.text} />;
```

Comment:
I still give you credit for not using only the index as key
setting our key to the index will forever be our last resort but we shall probably never do it
We are save to assume that users usernames will all be unique